### PR TITLE
Use GitHub API to tag and trigger a build instead

### LINKED ISSUE #48: Use Workflow Dispatch instead of a personal access token
I'm using a personal access token to tag and trigger a build if there's a new version of Cascadia. That's a pretty lame solution and it'd be much nicer to use Workflow Dispatch instead via this action https://github.com/marketplace/actions/workflow-dispatch

### PR DESCRIPTION
**Pull request by sitiom**
This way you don't need a personal access token anymore.

Resolves #48.

### CODE REVIEW

File: .github/workflows/check-for-new-cascadia.yml [hunk 1]
```diff
@@ -6,25 +6,45 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2        
+    - uses: octokit/request-action@v2.x
+      id: get_latest_release
```
Comment (adam7) [inline .github/workflows/check-for-new-cascadia.yml:12]: Is there a reason you're using an id here?

File: .github/workflows/check-for-new-cascadia.yml [hunk 2]
```diff
@@ -6,25 +6,45 @@ on:
 
 jobs:
   build:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-20.04
     steps:
-    - uses: actions/checkout@v2        
+    - uses: octokit/request-action@v2.x
```
Comment (adam7) [inline .github/workflows/check-for-new-cascadia.yml:11]: I like the way you've added names to the other steps, can you add to this too for consistency

File: .github/workflows/check-for-new-cascadia.yml [hunk 3]
```diff
@@ -6,25 +6,45 @@ on:
...
+        repo: cascadia-code
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Find the latest Cascadia tag
+      run: echo "CASCADIATAG=${{ fromJson(steps.get_latest_release.outputs.data).tag_name }}" >> $GITHUB_ENV
+    - name: Latest Cascadia tag is ${{ env.CASCADIATAG }}
+      run: echo "Latest Cascadia tag is ${{ env.CASCADIATAG }}"
+    - name: Check if tag exists
+      uses: mukunku/tag-exists-action@v1.0.0
+      id: check_tag
```
Comment (adam7) [inline .github/workflows/check-for-new-cascadia.yml:25]: This id seems unnecessary

File: .github/workflows/check-for-new-cascadia.yml [hunk 4]
```diff
@@ -6,25 +6,45 @@ on:
...
-          git config user.email github-actions@github.com
-          git tag -a ${CASCADIATAG} -m "Bump Cascadia version to ${CASCADIATAG}"    
-          git push origin --tags
-        fi  
+        route: GET /repos/{owner}/{repo}/releases/latest
+        owner: microsoft
+        repo: cascadia-code
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    - name: Find the latest Cascadia tag
```
Comment (adam7) [inline .github/workflows/check-for-new-cascadia.yml:19]: You could merge this step with the next one to get the latest tag and echo it in one step

**Review by adam7 — CHANGES_REQUESTED** [review]
Thanks for this @sitiom  it looks great. Just a couple of changes for simplicity and consistency and it's ready to merge.

File: .github/workflows/check-for-new-cascadia.yml (hunk 1 above)
Comment (sitiom) [reply .github/workflows/check-for-new-cascadia.yml:12]: The output of this step is needed to get the latest tag on the next step (`fromJson(steps.get_latest_release.outputs.data).tag_name`).

File: .github/workflows/check-for-new-cascadia.yml (hunk 3 above)
Comment (sitiom) [reply .github/workflows/check-for-new-cascadia.yml:25]: It is needed for the if condition on the next step (`if: ${{ steps.check_tag.outputs.exists != 'true' }}`)

File: .github/workflows/check-for-new-cascadia.yml (hunk 4 above)
Comment (sitiom) [reply .github/workflows/check-for-new-cascadia.yml:19]: Makes sense.

File: .github/workflows/check-for-new-cascadia.yml (hunk 1 above)
Comment (adam7) [reply .github/workflows/check-for-new-cascadia.yml:13]: Of course, my bad

**Review by adam7 — APPROVED** [review]

**Pull request merged**